lumigator/python/mzai/sdk/sdk/jobs.py

```diff
@@ -0,0 +1,135 @@
+"""Jobs SDK
+
+Provides a class to manipulate jobs in Lumigator.
+"""
+
+import asyncio
+from http import HTTPMethod
+from uuid import UUID
+
+import requests
+from schemas.extras import ListingResponse
+from schemas.jobs import (
+    JobCreate,
+    JobResponse,
+    JobResultDownloadResponse,
+    JobResultResponse,
+    JobType,
+)
+
+from sdk.client import ApiClient
+
+
+class Jobs:
+    JOBS_ROUTE = "jobs"
+
+    def __init__(self, c: ApiClient):
+        self.client = c
+
+    def get_jobs(self) -> ListingResponse[JobResponse]:
+        """Returns information on all jobs.
+
+        Returns:
+            ListingResponse[JobResponse]: all existing jobs.
+        """
+        response = self.client.get_response(self.JOBS_ROUTE)
+
+        if not response:
+            return []
+
+        return ListingResponse[JobResponse](**response.json())
+
+    def get_job(self, id: UUID) -> JobResponse:
+        """Returns information on a specific job.
+
+        Args:
+            id (UUID): the id of the job to retrieve
+        Returns:
+            JobResponse: the job information for the provided id.
+        """
+        response = self.client.get_response(f"{self.JOBS_ROUTE}/{id}")
+
+        if not response:
+            return []
+
+        return JobResponse(**(response.json()))
+
+    async def wait_for_job(self, id: UUID, retries: int = 30, poll_wait: int = 30) -> JobResponse:
+        """Waits for a job to succeed and returns its latest retrieved information.
+
+        Args:
+            id (UUID): the id of the job to wait for
+            retries (int):
+            poll_wait (int):
+        Returns:
+            JobResponse: the most recently job information for the id, when the job has finished
+        """
+        for _ in range(1, retries):
+            # http://localhost:8265/api/jobs/f311fa44-025a-4703-b8ba-7e0b1001b484
+            response = self.client.get_ray_job_response(f'{id}')
+            # response = requests.get(f"http://localhost:8265/api/jobs/{id}")
```

Comment:
Nit and non-blocking: I would remove these commented lines 🤫

Reply:
Done